src/pages/workspace/tags/WorkspaceTagsPage.tsx

```diff
@@ -79,6 +79,7 @@ function WorkspaceTagsPage({policyTags, route}: WorkspaceTagsPageProps) {
 
     useFocusEffect(
         useCallback(() => {
+            setSelectedTags({});
             fetchTags();
         }, [fetchTags]),
     );
```

Comment:
We should clear the tags when losing focus (instead of when gaining focus) that way we can remove the rest of logic that's used to clear the tags e.g. when navigating to the settings page

(same applies for the rest of the pages)